### ISSUE TITLE
Incorrect use of await and rAF in carousel

### ISSUE THREAD

**Issue by 20manas** [technojam/technojam.github.io#20]
#18
Since await is present at the beginning of __internet_step, the changes to css are **not** done before recalculations:
![screenshot_20190129_230316](https://user-images.githubusercontent.com/14952218/51928042-00581400-241b-11e9-87af-381fe0208239.png)
However if we move the await statement to before requestAnimationFrame and after toggling code.
Original:
![screenshot_20190129_230422](https://user-images.githubusercontent.com/14952218/51928367-c89d9c00-241b-11e9-8e96-f5472229e967.png)
New:
![screenshot_20190129_230503](https://user-images.githubusercontent.com/14952218/51928348-bf143400-241b-11e9-94c2-8b3e74e83002.png)
(ignore the pointer)
Then changes happen before recalculations, as they should:
![screenshot_20190129_230352](https://user-images.githubusercontent.com/14952218/51928517-25995200-241c-11e9-8200-c587175d7d71.png)
This is a minor issue but worth noting because if the beginning frame lags, then it might spiral down and cause lag on slower devices.

**Comment by 20manas** [comment]
There is one issue with the new code: The very first slide change happens immediately, but that can be taken care of with a simple `carmgr.start()` inside a `setTimeout`

**Comment by Prakhar-Srivastava** [comment]
Yeah, this is quite a serious and nasty bug, thanks for pointing it out.
Just completely slipped out of my head. I'll push the code tomorrow

**Comment by Prakhar-Srivastava** [comment]
Hey, I've tried moving await to the position you said, but the issue there is that the very first slide then changes almost instantaneously, and you've mentioned that we can use window.setTimeout with carmgr.start() but it doesn't work, thus the only solution which can be agreed upon is to use 
> await wait(3757)
> carmgr.start()
but doing this is a bit of overdo as I need to write 2 lines of code just to produce the same effect.
Also, I think the performance is barely affected by this as the GPU is rendering the transform: matrix3d(... code in the index.css's class .test_cls and if the first frame is delayed the CPU/GPU has 0.3s to recalculate and cache the process for the next render plus the delay between do consecutive window.requestAnimationFrame is enough to not jam the CPU/GPU.

**Comment by 20manas** [comment]
> Also, I think the performance is barely affected by this as the GPU is rendering the transform

Yes, this is why I said it is a minor issue. My intention was, either use rAF properly or don't use it at all as using it incorrectly like this could have undesired effects. I'll try to fix this myself if I can.
